test/js-api/runtimeexception/getArg.any.js

```diff
@@ -0,0 +1,58 @@
+// META: global=window,dedicatedworker,jsshell
+// META: script=/wasm/jsapi/memory/assertions.js
+
+test(() => {
+  const exn = new WebAssembly.Exception({ "parameters": [] })
+  const runtimeExn = new WebAssembly.RuntimeException(exn);
+  assert_throws_js(TypeError, () => runtimeExn.getArg());
+  assert_throws_js(TypeError, () => runtimeExn.getArg(exn));
+}, "Missing arguments");
+
+test(() => {
+  const invalidValues = [
+    undefined,
+    null,
+    true,
+    "",
+    Symbol(),
+    1,
+    {}
+  ];
+  const exn = new WebAssembly.Exception({ "parameters": [] })
+  const runtimeExn = new WebAssembly.RuntimeException(exn);
+  for (argument of invalidValues) {
+    assert_throws_js(TypeError, () => runtimeExn.getArg(argument, 0));
+  }
+}, "Invalid exception argument");
+
+test(() => {
+  const exn = new WebAssembly.Exception({ "parameters": [] })
+  const runtimeExn = new WebAssembly.RuntimeException(exn);
+  assert_throws_js(RangeError, () => runtimeExn.getArg(exn, 1));
+}, "Index out of bounds");
+
+test(() => {
+  const outOfRangeValues = [
+    undefined,
+    NaN,
+    Infinity,
+    -Infinity,
+    -1,
+    0x100000000,
+    0x1000000000,
+    "0x100000000",
+    { valueOf() { return 0x100000000; } },
+  ];
+
+  const exn = new WebAssembly.Exception({ "parameters": [] })
+  const runtimeExn = new WebAssembly.RuntimeException(exn);
+  for (const value of outOfRangeValues) {
+    assert_throws_js(TypeError, () => runtimeExn.getArg(exn, value));
+  }
+}, "Getting out-of-range argument");
+
+test(() => {
+  const exn = new WebAssembly.Exception({ "parameters": ["i32"] })
+  const runtimeExn = new WebAssembly.RuntimeException(exn, 42);
```

Comment:
#154 says the second argument should be a `sequence` (i.e., an array-like), so this should throw. Do we want variadic arguments instead?

Reply:
Thanks for catching this, I had implemented this wrong in my prototype with variadic args and it leaked into the tests. I think it makes more sense to use a sequence argument here as the spec draft says.

(should be fixed in latest commit)